src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -13,8 +13,12 @@
 
 class AdalAuthentication(Authentication):  # pylint: disable=too-few-public-methods
 
-    def __init__(self, token_retriever):
-        self._token_retriever = token_retriever
+    def __init__(self, token):
+        self.token = token
```

Comment:
We should not do this. We like to use a token with guaranteed validity, so this is why we acquire one right before sending out the request on the wire. Some commands take a long time to run, say vmss with 100 instances, the change here will break it because the same token used in the subsequent polling might get expired.